### ISSUE TITLE
iOS 9 Invalid attempt to spread non-iterable instance

### ISSUE THREAD

**Issue by ym-is-great** [fengyuanchen/compressorjs#60]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x ] Bug report 
[ ] Feature request
[ ] Documentation issue or request
</code></pre>

## Current behavior

iOS9 设备上的 Safari 始终报 “Invalid attempt to spread non-iterable instance” 错误

调试后发现 `_arrayWithoutHoles` 方法中的 `Array.isArray()` 返回 false，移除这个判断后不报错了，怀疑是 iOS9 的把数组误判断为不是数组。

改成这样后功能正常了:

```
function _toConsumableArray(arr) {
  return _iterableToArray(arr) || _arrayWithoutHoles(arr) || _nonIterableSpread();
}

function _arrayWithoutHoles(arr) {
  for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) arr2[i] = arr[i];

  return arr2;
}
```

## Environment

<pre><code>
Compressor.js version: X.Y.Z

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [x] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

Others:

</code></pre>

**Comment by fengyuanchen** [comment]
What's the Safari version in your iOS 9 system?

**Comment by ym-is-great** [comment]
Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13E238 Safari/601.1

> What's the Safari version in your iOS 9 system?

**Comment by fengyuanchen** [comment]
It seems Safari 9 on iOS 9.3 does not support [`Symbol.iterable in Typed Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array#Browser_compatibility)

**Comment by fengyuanchen** [comment]
I don't have an iOS 9 device for testing, can you print this line in your device and tell me the result (`true` or `false`)?

```js
window.alert(Symbol.iterator in Object(new Uint8Array([1, 2, 3])));
```

**Comment by fengyuanchen** [comment]
Or try the following helpers:

```js
function _toConsumableArray(arr) {
  return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _nonIterableSpread();
}

function _arrayWithoutHoles(arr) {
  if (Array.isArray(arr)) {
    for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) arr2[i] = arr[i];

    return arr2;
  }
}

function _iterableToArray(iter) {
  if (Symbol.iterator in Object(iter) || /^\[object (?:Arguments|(?:Int|Uint|Float)(?:8|16|32|64)Array)\]$/.test(Object.prototype.toString.call(iter)) return Array.from(iter);
}

function _nonIterableSpread() {
  throw new TypeError("Invalid attempt to spread non-iterable instance");
}
```

**Comment by edi** [comment]
I have a BrowserStack subscription, figured it can be useful.

``` Safari 9 on iOS 9 - iPhone 6S
Symbol.iterator in Object(new Uint8Array([1, 2, 3])) // => false```

As for the other helpers, you seem to be missing a closing parenthesis before return, in the if statement.

```
toString.call(iter)) ")" return Array.
```

Other than that, I've defined them, and ran the following with no errors:
```
var arr = [1,2,'3']; // not sure what input you want me to check
_toConsumableArray( arr ); //=> []
```

Whereas, on my Chrome on Mac ( latest version at the moment of posting ) it returns the array as it was inputted. `_toConsumableArray(a) //=> (3) [1, 2, "3"]`. On the other hand, testing the actual tool seems to be working just fine ( on the device mentioned above ) as it compresses and returns the  output correctly.

Hope it helps, let me know if you want me to test anything else.

**Comment by fengyuanchen** [comment]
@EduardJS I had fixed this issue as of v1.0.3.
